### PR TITLE
feat: ACNA-3097 - support web:raw

### LINKED ISSUE #79: Enable web:raw on local "aio app dev"
Running "aio app dev" currently does not support "web: raw" yet, so I could not test receiving the multi-part data in params.__ow_body (it's empty). For now, I tested in stag Runtime which worked fine.
Would you please add the support for "web:raw" in local dev environment as well?

### PR DESCRIPTION
**Pull request by shazron**
fixes #79 

## Description

The POST behavior below has been verified on adobeioruntime.net, and the dev server matches the behavior.
This has been verified by the e2e tests.

There is no spec, just [example docs](https://github.com/apache/openwhisk/blob/master/docs/webactions.md#http-context), and the behavior is specified from the docs: `The __ow_body property is present either when handling "raw" HTTP requests, or when the HTTP request entity is not a JSON object or form data`

Note the quirk in the table for `application/x-www-form-urlencoded, raw`.

| content-type                                        | non-raw                          | raw                                                 |
|-------------------------------------|------------------------|---------------------------------|
| application/json                                    | promoted to params     | __ow_body base64 (stringified) |
| application/x-www-form-urlencoded | promoted to params     | __ow_body set to query string   |
| multipart/form-data                              | __ow_body base64       | __ow_body base64                    |
| text/plain                                                | __ow_body set to value | __ow_body set to value            |

## How Has This Been Tested?

- npm test
- npm run e2e (local)
- npm run e2e (production)

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the **CONTRIBUTING** document.
- [X] I have added tests to cover my changes.
- [X] All new and existing tests passed.

### CODE REVIEW

**Review by sandeep-paliwal — APPROVED** [review]
LGTM

File: src/lib/run-dev.js [hunk 1]
```diff
@@ -552,11 +553,27 @@ function createActionParametersFromRequest ({ req, actionInputs = {} }) {
...
+  const isJson = req.is('application/json')
+  const isFormData = req.is('application/x-www-form-urlencoded')
+  const isRaw = isRawWebAction(contextItem)
+
+  if (params.__ow_method === 'post' && req.body !== null) {
+    if (isRaw) {
+      if (isFormData) {
+        params.__ow_body = new URLSearchParams(req.body).toString() // convert json back to query string
+      } else if (isJson) {
+        params.__ow_body = Buffer.from(JSON.stringify(req.body)).toString('base64')
```
Comment (moritzraho) [inline src/lib/run-dev.js:566]: nit: oportunity to merge the two `else if / else` branch? as utils.bodyTransformToRaw also supports objects right?

**Review by moritzraho — APPROVED** [review]
clean!

File: src/lib/run-dev.js [hunk 2]
```diff
@@ -160,8 +162,11 @@ async function runDev (runOptions, config, _inprocHookRunner) {
   }
 
   const app = express()
+  app.use(express.text({ type: 'text/plain' }))
+  app.use(express.json({ strict: false }))
```
Comment (hparra) [inline src/lib/run-dev.js:166]: Consider adding:

- `inflate: false`: express will decompress automatically but OpenWhisk does not so we should not give devs impression that just adding "Content-Type: gzip" will work.
- `limit: '1MB'`: to mimic OpenWhisk limitations

Comment (shazron) [reply src/lib/run-dev.js:166]: thanks! will add those

File: src/lib/run-dev.js (hunk 1 above)
Comment (shazron) [reply src/lib/run-dev.js:566]: thanks! will update

File: src/lib/run-dev.js (hunk 2 above)
Comment (shazron) [reply src/lib/run-dev.js]: fixed in 76c4bc698d399481285846f55ec01bbfe2f72243

File: src/lib/run-dev.js (hunk 1 above)
Comment (shazron) [reply src/lib/run-dev.js]: fixed in 8893668532b89625c367622850bf00450e2d3b62

**Pull request merged**